data/src/main/java/com/architectcoders/aacboard/data/repository/RegionRepositoryImpl.kt

```diff
@@ -0,0 +1,30 @@
+package com.architectcoders.aacboard.data.repository
+
+import com.architectcoders.aacboard.data.datasource.local.DeviceDataSource
+import com.architectcoders.aacboard.domain.repository.RegionRepository
+
+class RegionRepositoryImpl(
+    private val deviceDataSource: DeviceDataSource,
+    private val permissionChecker: PermissionChecker
+) : RegionRepository {
+
+    override suspend fun getLastUserRegion(): String {
+        return if (permissionChecker.check(PermissionChecker.Permission.COARSE_LOCATION)) {
+            deviceDataSource.getLastUserRegion()?.lowercase() ?: DEFAULT_USER_REGION
+        } else {
+            DEFAULT_USER_REGION
+        }
+    }
+
+
+    companion object {
+        private const val DEFAULT_USER_REGION = "us"
+    }
+}
+
+interface PermissionChecker {
```

Comment:
I would separate the PermissionChecker Interface to another class, it doesn't need to be related to location permissions only. It may grow with other permissions checks.

Reply:
Yeah, you're right, good catch!